src/components/Controls.vue

```diff
@@ -82,11 +81,11 @@ export default {
       );
     }
     this.graphStructure = await loadData(workspace, graph, host);
-    const { provenance, app } = setUpProvenance(this.graphStructure.nodes);
-    this.app = app;
-    this.provenance = provenance;
+    this.provenance = setUpProvenance(this.graphStructure);
     this.workspace = workspace;
     this.graph = graph;
+
+    document.addEventListener('keydown', this.keyDownHandler);
```

Comment:
Can you look to see if there's some "Vueier" way of binding global events to actions within a component? If not, this is fine. But it feels like Vue probably has a way to do this that might be a little less bare metal than using `addEventListener`.

Also, there's a nice-loooking library called [mousetrap](https://craig.is/killing/mice) that you might want to try out here. (You might not, seeing as you have a nice short implementation for keybindings here already, but just passing it along.)

Reply:
It's easy to bind key combos to certain DOM elements, but globally, it's more tricky. Here are a few things I found:
- [How to capture keystrokes, vue forum](https://forum.vuejs.org/t/capture-keypress-for-all-keys/14560)
- [Using raw js instead of vue-shortcut](https://shubhamjain.co/til/vue-shortcuts/)
- [Using ctrl-v in an app, bind to 1 input](https://stackoverflow.com/questions/42711537/vuejs-how-to-detect-ctrlv)
    - > <input @keyup.ctrl.76="YourFn">

Mousetrap could be cool, but I think it adds a unnecessary dependency and we have a good implementation here already

Reply:
Thanks for looking into it. What you've got looks like the best way to do it.